### PR TITLE
Fix README.adoc text about supported versions

### LINKED ISSUE #2178: Incorrect information about supported languages in readme
Readme says "Supports multiple languages and frameworks such as Node.js, Java, Ruby, Perl, PHP, and Python." But odo officially supports only Java and NodeJs, the other languages are without any guarantees. 

Not sure what the right wording should be.

### PR DESCRIPTION
**Pull request by pmacko1**
**What kind of PR is this?**

/kind docs

**What does does this PR do / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #2178 #2093

**How to test changes / Special notes to the reviewer**:

### CODE REVIEW

File: README.adoc [hunk 1]
```diff
@@ -24,11 +24,12 @@ Existing tools such as `oc` are more operations-focused and require a deep-under
...
 `odo` is designed to be simple and concise with the following key features:
 
-* Simple syntax and design centered around concepts familiar to developers, such as project, application, and component.
+* Simple syntax and design centered around concepts familiar to developers, such as projects, applications, and components.
 * Completely client based. No server is required within the OpenShift cluster for deployment.
-* Supports multiple languages and frameworks such as Node.js, Java, Ruby, Perl, PHP, and Python.
-* Detects changes to local code and deploys it to the cluster automatically, giving instant feedback to validate changes in real-time.
-* Lists all available components and services from the OpenShift cluster.
+* Official full support of Node.js and Java components.
+* Partial compatibility with many languages and frameworks such as Ruby, Perl, PHP, and Python. Run odo catalog list components to see the list of supported and unsupported components.
```
Comment (boczkowska) [inline README.adoc]: Run odo catalog list components to see -> Run `odo catalog list components` to see

Comment (boczkowska) [reply README.adoc]: And remove "and unsupported components".

**Comment by boczkowska** [comment]
@pmacko1 you can refer for tomas' https://github.com/openshift/odo/issues/2093#issuecomment-528777633 to get a list of supported images.

File: README.adoc (hunk 1 above)
Comment (Preeticp) [reply README.adoc]: Run odo catalog list components to see... needs to e removed from this conceptual module. It should be part of a procedural module.

**Comment by kadel** [comment]
> @kadel I am not sure but I vaguely remember a discussion/issue that talked about being more precise than just saying that odo supports Java because Java could mean a lot of things. By things I mean frameworks, versions, etc. not the coffee, of course. 😉

Technically we support any generic Java (OpenJDK) projects

File: README.adoc (hunk 1 above)
Comment (pmacko1) [reply README.adoc]: according to @kadel `odo catalog list components` can have different outputs depending on the cluster's configuration. I'd remove this command altogether from here and find a new place to put it

**Comment by pmacko1** [comment]
@kadel @amitkrout Hello guys, can I ask you for a review, please?

Thanks!

**Comment by pmacko1** [comment]
@boczkowska or @Preeticp can I ask one of you for a review, please? :)

**Comment by Preeticp** [comment]
@cdrage, @amitkrout  can you please take a look at this PR. 

Questions: 
1. Considering that we are adding a section telling the users about the supported languages and images right under the Key features sections, do we still want to add these points in the key features:
"- Official full support of Node.js and Java components.
 - Partial compatibility with languages and frameworks such as Ruby, Perl, PHP, and Python.
"
2. Also, is the first point technically correct? I remember Steve S pointing out that we do not support all the Java images. So would 'full support' imply the right thing to the user? Would it be safer if we simply remove these 2 points and just have the Supported languages and images section as done in this PR.

**Comment by amitkrout** [comment]
> @cdrage, @amitkrout can you please take a look at this PR.
> 
> Question: Also, is the first point technically correct? I remember Steve S pointing out that we do not support all the Java images. So would 'full support' imply the right thing to the user? Would it be safer if we simply remove these 2 points and just have the Supported languages and images section as done in this PR.

AFAIK java image stream contains these supported container image like ```openjdk/openjdk-11-rhel7, ...``` and these container images are used when a java component is created and pushed. If some body wants to apply his/her won java image stream, then i am not sure in which category it falls in ```odo catalog list components```. More importantly we can not guaranty customize java or nodejs image stream having container image not same as we test on.

IMO instead of writing ```... full support ...``` i would recommend just remove ```full``` from the sentence.  @cdrage WDYT ?

**Comment by pmacko1** [comment]
WIP, as the description of the command is not correct. I received an update from Tomas.

File: README.adoc [hunk 2]
```diff
@@ -24,11 +24,26 @@ Existing tools such as `oc` are more operations-focused and require a deep-under
...
-* Lists all available components and services from the OpenShift cluster.
+* Official full support of Node.js and Java components.
+* Partial compatibility with languages and frameworks such as Ruby, Perl, PHP, and Python. 
+* Detects changes to local code and deploys it to the cluster automatically, giving instant feedback to validate changes in real time.
+* Lists all the available components and services from the OpenShift cluster.
+
+=== Supported languages and corresponding container images
+Run `odo catalog list components` to list the supported languages and corresponding container images present on your cluster.
+
+.Supported container images
```
Comment (cdrage) [inline README.adoc]: Maybe `Officially supported container images` ?

File: README.adoc [hunk 3]
```diff
@@ -24,11 +24,26 @@ Existing tools such as `oc` are more operations-focused and require a deep-under
 
 `odo` is designed to be simple and concise with the following key features:
 
-* Simple syntax and design centered around concepts familiar to developers, such as project, application, and component.
+* Simple syntax and design centered around concepts familiar to developers, such as projects, applications, and components.
 * Completely client based. No server is required within the OpenShift cluster for deployment.
-* Supports multiple languages and frameworks such as Node.js, Java, Ruby, Perl, PHP, and Python.
-* Detects changes to local code and deploys it to the cluster automatically, giving instant feedback to validate changes in real-time.
-* Lists all available components and services from the OpenShift cluster.
+* Official full support of Node.js and Java components.
```
Comment (cdrage) [inline README.adoc]: remove the `full` ?

File: README.adoc [hunk 4]
```diff
@@ -24,11 +24,26 @@ Existing tools such as `oc` are more operations-focused and require a deep-under
...
+
+.Supported container images
+[options="header"]
+|====
+|Node.js|Java
+|`centos/nodejs-8-centos7`|`redhat-openjdk-18/openjdk18-openshift`
+|`rhoar-nodejs/nodejs-8`|`openjdk/openjdk-11-rhel8`
+|`rhoar-nodejs/nodejs-10`|`openjdk/openjdk-11-rhel7`
+|`bucharestgold/centos7-s2i-nodejs`|
+|`rhscl/nodejs-8-rhel7`|
```
Comment (amitkrout) [inline README.adoc]: Also include image ```rhscl/nodejs-10-rhel7```.

Reference - https://github.com/openshift/odo-init-image/pull/43/commits/0ce478aefc8f46813ec0b94187e537ec5d0a40b4#diff-23c2abb47c564635a127893860b3cb08R6

File: README.adoc (hunk 2 above)
Comment (pmacko1) [reply README.adoc]: might be a little redundant, maybe. We have "compatibility" vs "support" there now.

On a second thought, we might want to include unsupported languages there as a new table as well. But explicitly state, that no support/compatiblity is guaranteed. WDYT?

File: README.adoc (hunk 3 above)
Comment (pmacko1) [reply README.adoc]: yep

File: README.adoc (hunk 4 above)
Comment (pmacko1) [reply README.adoc]: :+1:

**Comment by pmacko1** [comment]
@amitkrout would `Official support of Node.js and Java components based on Red Hat images.` be better worded?

**Pull request merged**